Security/Core/Authentication/Provider/Provider.php

```diff
@@ -158,7 +162,7 @@ protected function getCurrentTime()
 
     protected function isTokenFromFuture($created)
     {
-        return strtotime($created) > strtotime($this->getCurrentTime());
+        return strtotime($created) - $this->clockSkew > strtotime($this->getCurrentTime());

```

Comment:
If we take clock skew (offset) into consideration upon determination whether the token is from the future (by subtracting the skew from the created time), shouldn't we also take the same clock skew (offset) into consideration when determining whether the token is valid (by adding the skew to the created time)?

Perhaps extracting https://github.com/hostage-nl/EscapeWSSEAuthenticationBundle/blob/0b1c36e8151049aba7beeaa27045dd19ca30d65d/Security/Core/Authentication/Provider/Provider.php#L125-L129 into a isTokenExpired-method, where clock skew is used as follows?
`strtotime($this->getCurrentTime()) - (strtotime($created) + $this->clockSkew) > $this->lifetime`